netaddr/ip/__init__.py

```diff
@@ -1683,28 +1683,33 @@ def spanning_cidr(ip_addrs):
 
     :return: a single spanning `IPNetwork` subnet.
     """
-    sorted_ips = sorted(
-        [IPNetwork(ip) for ip in ip_addrs])
-
-    if not len(sorted_ips) > 1:
+    ip_addrs_count = 0
+    min_network = None
+    max_network = None
+    for ip in ip_addrs:
+        ip_addrs_count += 1
+        network = IPNetwork(ip)
+        if min_network is None or network < min_network:
```

Comment:
I could have used `min()` and `max()`, but since the point of the change was performance, seemed silly to make it 2N instead of N. If you think the complexity of the code isn't worth the extra pass I can change it to use `min()` and `max()`.